packages/fxa-auth-server/lib/payments/stripe.js

```diff
@@ -166,6 +173,134 @@ class StripeHelper {
     return customer;
   }
 
+  /**
+   * Fetch a customer for the record from Stripe based on user ID & email.
+   *
+   * Uses Redis caching if configured.
+   *
+   * @param {string} uid Firefox Account Uid
+   * @param {string} email Firefox Account Email
+   * @returns {Promise<Customer|void>} Customer if exists in the system.
+   */
+  async customer(uid, email) {
+    const cacheKey = this.customerCacheKey(uid, email);
+    return cachedResult(
+      this.log,
+      this.redis,
+      cacheKey,
+      this.cacheTtlSeconds,
+      async () => {
+        const customer = await this.fetchCustomer(uid, email, [
+          'data.sources',
+          'data.subscriptions',
+        ]);
+        if (customer) {
+          this.cachePropsForCustomerId(customer.id, { uid, email });
+        }
+        return customer;
+      }
+    );
+  }
+
+  /**
+   * Fetch a subscription for a customer from Stripe.
+   *
+   * Uses Redis caching if configured.
+   *
+   * @param {string} uid Firefox Account Uid
+   * @param {string} email Firefox Account Email
+   * @param {string} subscriptionId Subscription ID
+   * @returns {Promise<Subscription|void>} Customer if exists in the system.
+   */
+  async subscriptionForCustomer(uid, email, subscriptionId) {
+    const customer = await this.customer(uid, email);
+    return customer.subscriptions.data.find(
+      subscription => subscription.id === subscriptionId
+    );
+  }
+
+  /**
+   * Delete a cached customer record based on user ID & email.
+   *
+   * @param {string} uid Firefox Account Uid
+   * @param {string} email Firefox Account Email
+   * @returns {Promise<void>} Redis result.
+   */
+  async deleteCachedCustomer(uid, email) {
+    try {
+      return deleteCachedResult(
+        this.log,
+        this.redis,
+        this.customerCacheKey(uid, email)
+      );
+    } catch (err) {
+      this.log.error(`subhub.deleteCachedCustomer.failed`, { err });
+    }
+  }
+
+  /**
+   * Delete a cached customer record based on customer ID.
+   *
+   * @param {string} customerId A Stripe customer ID
+   * @returns {Promise<void>} Redis result.
+   */
+  async deleteCachedCustomerById(customerId) {
```

Comment:
I believe this is the result of me saying that the `customer.subscription.*` events doesn't have the necessary components (uid, email) to construct the cache key.  However, we are no longer listening to those events since a subscription change also triggers "customer.update".

Ignore me if I'm mistaken!

Reply:
Ah, yup, that's what these bits are for. If we don't need them, I can back them out